### PR TITLE
Use appropriate ACL definition based on the channel.

### PR DESCRIPTION
**Pull request by frankban**
Use development ACLs if the development entity is requested.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1274/
Test PASSed.

File: internal/v4/auth.go [hunk 1]
```diff
@@ -121,14 +121,22 @@ func (h *ReqHandler) checkRequest(req *http.Request, entityId *router.ResolvedUR
 // AuthorizeEntity checks that the given HTTP request
 // can access the entity with the given id.
 func (h *ReqHandler) AuthorizeEntity(id *router.ResolvedURL, req *http.Request) error {
-	baseEntity, err := h.Store.FindBaseEntity(&id.URL, "acls")
+	field := "acls"
```
Comment (mhilton) [inline internal/v4/auth.go]: I wonder if this optimization is worth it. perhaps just fetch both fields?

**Comment by mhilton** [comment]
:+1:

File: internal/v4/auth.go (hunk 1 above)
Comment (frankban) [inline internal/v4/auth.go]: Yes, why not.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1275/
Test PASSed.

**Comment by urosj** [comment]
:+1:

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**